extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java

```diff
@@ -216,7 +215,9 @@ SyntheticBeanBuildItem initBasicAuth(
                 && !buildTimeConfig.auth.basic.orElse(false)) {
             //if not explicitly enabled we make this a default bean, so it is the fallback if nothing else is defined
             configurator.defaultBean();
-            securityInformationProducer.produce(SecurityInformationBuildItem.BASIC());
+            if (buildTimeConfig.auth.basic.isPresent() && buildTimeConfig.auth.basic.get()) {
```

Comment:
line 215 `!buildTimeConfig.auth.basic.orElse(false)` means that

`quarkus.http.auth.basic` is missing => inside IF body
`quarkus.http.auth.basic=false` => inside IF body
`quarkus.http.auth.basic=true` => don't go in

while line 218 says

`buildTimeConfig.auth.basic.isPresent()` => `quarkus.http.auth.basic=false` || `quarkus.http.auth.basic=true`
`buildTimeConfig.auth.basic.get()` => `quarkus.http.auth.basic=true`

so in order to have produced `BASIC` here it means that `quarkus.http.auth.basic` must be both `true` and `false`

@phillip-kruger I don't know what was intention of this condition, can you please fix it? Thank you